src/proxy.rs

```diff
@@ -0,0 +1,133 @@
+//! Client Proxy support.
+use std::net::TcpStream;
+use std::io;
+
+use Client;
+use net::{HttpStream, NetworkConnector};
+use http::h1::Http11Protocol;
+use method::Method;
+use std::fmt;
+use url::Url;
+
+use header::ProxyConnection;
+
+
+#[derive(Copy)]
+pub enum ProxyPolicy {
+    /// Proxy all requests.
+    ProxyAll,
+    /// Proxy only http requests.
+    ProxyHttp,
+    /// Proxy only https requestss.
+    ProxyHttps,
+    /// Proxy if the contained function returns true.
+    ProxyIf(fn(&Url) -> bool),
+}
+
+// Need to implement Clone for the time being.
+impl Clone for ProxyPolicy {
```

Comment:
This is due to a bug in rustc, because of the function pointer in `ProxyIf`? Is there a issue number for rustc?